src/engine.rs

```diff
@@ -104,18 +104,27 @@ impl Engine {
             let epv = end_point_voices.clone();
 
             let sounds = Arc::new(Mutex::new(Vec::new()));
-            future_to_exec = Some(stream.for_each(move |mut buffer| -> Result<_, ()> {
+            future_to_exec = Some(stream.for_each(move |mut buffer| -> Result<(), ()> {
                 let mut sounds = sounds.lock().unwrap();
 
                 {
                     let mut pending = epv.pending_sounds.lock().unwrap();
                     sounds.append(&mut pending);
                 }
 
+
                 if sounds.len() == 0 {
                     return Ok(());
                 }
                 sounds.retain(|s| !s.1.local_dead);
+                //Drop if it's not playing a real source, and it's sink is detached
+                sounds.retain(|s|{
+                    if !s.1.is_playing_real_source{
+                        Arc::strong_count(&s.1.next)>1
```

Comment:
Also, that logic needed to be inverted.  This was the cause of the issue you mentioned:

Edit: Not sure why but this seems to be preventing the first sound on an endpoint from playing, so it's probably not ready to be merged yet.